nodepool/elements/jenkins-slave/pre-install.d/20-add-openjdk-ppa-to-trusty

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+if [ ${DIB_DEBUG_TRACE:-0} -gt 0 ]; then
+    set -x
+fi
+set -eu
+set -o pipefail
+
+source /etc/lsb-release
+if [[ "${DISTRIB_CODENAME}" == "trusty" ]]; then
+  add-apt-repository -y ppa:openjdk-r/ppa
```

Comment:
It doesn't, but diskimage-builder does. Well, it seems to, because this definitely works - I've verified in my test environment.

Reply:
Do we need to run update-java-alternatives to set the default JRE?  IIRC something in my testing failed when I installed openjdk-8-jre-headless rather than default-jre-headless because some default symlink hadn't been created.

In https://github.com/rcbops/rpc-gating/pull/681, I installed both JREs (7 and 8) on trusty, but after the jenkins agent started I flipped the default back to 7 so that anything using the instance wouldn't accidentally make reliance on a 3rd party package.  Is doing something similar here possible w/ nodepool?

Reply:
Confirmed that diskimage-builder does it after the pre-install.d stage:

```
2018-04-13 12:55:32.757 | dib-run-parts 04-dib-init-system completed
2018-04-13 12:55:32.757 | dib-run-parts Running /tmp/in_target.d/pre-install.d/20-add-openjdk-ppa-to-trusty
2018-04-13 12:55:32.760 | + set -eu
2018-04-13 12:55:32.760 | + set -o pipefail
2018-04-13 12:55:32.760 | + source /etc/lsb-release
2018-04-13 12:55:32.760 | ++ DISTRIB_ID=Ubuntu
2018-04-13 12:55:32.760 | ++ DISTRIB_RELEASE=14.04
2018-04-13 12:55:32.760 | ++ DISTRIB_CODENAME=trusty
2018-04-13 12:55:32.760 | ++ DISTRIB_DESCRIPTION='Ubuntu 14.04.5 LTS'
2018-04-13 12:55:32.760 | + [[ trusty == \t\r\u\s\t\y ]]
2018-04-13 12:55:32.760 | + add-apt-repository -y ppa:openjdk-r/ppa
2018-04-13 12:55:33.881 | gpg: keyring `/tmp/tmp5_kyghwg/secring.gpg' created
2018-04-13 12:55:33.882 | gpg: keyring `/tmp/tmp5_kyghwg/pubring.gpg' created
2018-04-13 12:55:33.882 | gpg: requesting key 86F44E2A from hkp server keyserver.ubuntu.com
2018-04-13 12:55:34.117 | gpg: /tmp/tmp5_kyghwg/trustdb.gpg: trustdb created
2018-04-13 12:55:34.117 | gpg: key 86F44E2A: public key "Launchpad OpenJDK builds (all archs)" imported
2018-04-13 12:55:34.117 | gpg: Total number processed: 1
2018-04-13 12:55:34.117 | gpg:               imported: 1  (RSA: 1)
2018-04-13 12:55:34.175 | OK
2018-04-13 12:55:34.202 | dib-run-parts 20-add-openjdk-ppa-to-trusty completed
2018-04-13 12:55:34.203 | dib-run-parts Running /tmp/in_target.d/pre-install.d/99-apt-get-update
2018-04-13 12:55:34.205 | + set -eu
2018-04-13 12:55:34.205 | + set -o pipefail
2018-04-13 12:55:34.205 | + apt-get -y update
2018-04-13 12:55:34.389 | Get:1 http://ppa.launchpad.net trusty InRelease [15.4 kB]
2018-04-13 12:55:34.558 | Get:2 http://ppa.launchpad.net trusty/main amd64 Packages [4672 B]
2018-04-13 12:55:34.606 | Ign https://mirror.rackspace.com trusty InRelease
...
```

Reply:
@mattt416 We can't really do the second part, because jenkins doesn't interact with the image build process at all. So if there's an issue with the presented instance, the test should work that out. I think this is better anyway, as it's entirely plausible for a customer to have java installed already and if we rely in a product on another version, it's best to ensure we work that stuff out in the deployment tooling.

For the first question - good one. I have no idea. I can use this to update the live nodes and see whether jenkins connection to them.

Reply:
Thanks @odyssey4me -- regarding #2, if we merge w/ this then I'll update the other ansible playbook to do the same thing so we're consistent across builds.

Reply:
Proved in https://rpc.jenkins.cit.rackspace.net/job/RE-unit-test-slave-types/1180 that using a node with this new element change works just fine.